backend/server/routes/establishments/workerCertificate/workerCertificateService.js

```diff
@@ -0,0 +1,206 @@
+    if (!files || !files.length) {
+      throw new HttpError('Missing `files` param in request body', 400);
+    }
+
+    if (!files.every((file) => file?.filename)) {
+      throw new HttpError('Missing file name in request body', 400);
+    }
+
+    const responsePayload = [];
+
+    for (const file of files) {
+      const filename = file.filename;
+      const fileId = uuidv4();
+      const key = this.makeFileKey(establishmentUid, workerUid, recordUid, fileId);
+      const signedUrl = await s3.getSignedUrlForUpload({
+        bucket: certificateBucket,
+        key,
+        options: { expiresIn: uploadSignedUrlExpire },
+      });
+      responsePayload.push({ filename, signedUrl, fileId, key });
+    }
+
+    return responsePayload;
+  };
+
+  async confirmUpload(req) {
+    const { files } = req;
+    const { establishmentUid, recordUid } = req.params;
+
+    if (!files || !files.length) {
+      throw new HttpError('Missing `files` param in request body', 400);
+    }
+
+    const record = await this.certificateTypeModel.findOne({
+      where: {
+        uid: recordUid,
+      },
+      attributes: ['id', 'workerFk'],
+    });
+
+    if (!record) {
+      throw new HttpError(`Failed to find related ${this.recordType} record`, 400);
+    }
+
+    const { workerFk, id: recordId } = record.dataValues;
+
+    const etagsMatchRecord = await this.verifyEtagsForAllFiles(files);
+
+    if (!etagsMatchRecord) {
+      throw new HttpError('Failed to verify files on S3', 400);
+    }
+
+    for (const file of files) {
+      const { filename, fileId, key } = file;
+
+      try {
+        await this.certificatesModel.addCertificate({ recordId, workerFk, filename, fileId, key });
+      } catch (err) {
+        console.log(err);
+        throw new HttpError('Failed to add records to database', 500);
+      }
+    }
+  };
+
+  verifyEtagsForAllFiles = async (files) => {
+    try {
+      for (const file of files) {
+        const etagMatchS3Record = await s3.verifyEtag(certificateBucket, file.key, file.etag);
+        if (!etagMatchS3Record) {
+          console.error('Etags in the request does not match the record at AWS bucket');
+          return false;
+        }
+      }
+    } catch (err) {
+      console.error(err);
+      return false;
+    }
+    return true;
+  };
+
+  async getPresignedUrlForCertificateDownload ({files, params}) {
+    const { establishmentUid, workerUid, recordUid } = params;
+
+    if (!files || !files.length) {
+      throw new HttpError('No files provided in request body', 400);
+    }
+
+    const responsePayload = [];
+
+    for (const file of files) {
+      const signedUrl = await s3.getSignedUrlForDownload({
+        bucket: certificateBucket,
+        key: this.makeFileKey(establishmentUid, workerUid, recordUid, file.uid),
+        options: { expiresIn: downloadSignedUrlExpire },
+      });
+      responsePayload.push({ signedUrl, filename: file.filename });
+    }
+
+    return { files: responsePayload };
```

Comment:
```suggestion
    return responsePayload;
```
the endpoint controller have `res.status(200).json({ files: responsePayload })` so we don't need to wrap it here